### PR TITLE
refactor: Remove bip39 library

### PR DESCRIPTION
**Pull request by sleepdefic1t**
## Proposed changes

bip39-lib has several bugs as discussed in #81 & #72 

This PR removes the mnemonic class, tests, documentation, examples, and lib dependency (bip39).

It is also a breaking change and bumps C++ Crypto to v1.0.0

## Types of changes

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes

### CODE REVIEW

**Comment by sleepdefic1t** [comment]
Looks like there are additional build issues with ESP-8266 with PlatformIO on Linux-only.

I can confirm that code builds and flashes to an ESP8266, apparently just not on Linux atm.

Will investigate more and decide what to do with this PR.

[WIP] for now.

**Comment by faustbrian** [comment]
`1.0.0` is reserved for the first stable release of all crypto packages when AIP11 is implemented in all of them.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,19 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 - removed bip39 and mnemonic feature ([#86])
 
+## [0.4.0] - 2019-05-20
+
+### Changed
+
+- updated vendorField to support 255 bytes in Core v2.4 ([#84])
```
Comment (faustbrian) [inline CHANGELOG.md]: When you add those numbers in the brackets you also need to add them here https://github.com/ArkEcosystem/cpp-crypto/pull/86/files/90bea179a4eb262800b98fad60064d0a616132a2..dc7034f4fbafb49c828996cbc2031cc24bdcead9#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR92.

**Comment by sleepdefic1t** [comment]
@faustbrian 
Should be good to go now.
Thank you.

**Comment by sleepdefic1t** [comment]
@faustbrian 

This is a breaking change, but it’s ready.
Can we merge with a minor bump,
or are we waiting until 1.0?

**Comment by faustbrian** [comment]
We can release this with a minor bump, everything before `1.0.0` should be seen as unstable and can have breaking changes any time.

**Comment by sleepdefic1t** [comment]
🤔 
Nothing in this PR should affect that failing CI test in `build-linux-clang-5`.
#92 passed with the the same transaction class and tests ([circleci/ArkEcosystem/cpp-crypto/713](https://circleci.com/gh/ArkEcosystem/cpp-crypto/713)).

You can rebuild if you want to double-check, @faustbrian.
Otherwise, this should be good to go 👍

**Pull request merged**